app/controllers/energy_tariffs/energy_tariffs_controller.rb

```diff
@@ -55,6 +55,35 @@ def choose_meters
       end
     end
 
+    def edit_meters
+      if @energy_tariff.electricity?
+        @meters = @tariff_holder.meters.electricity
+      elsif @energy_tariff.gas?
+        @meters = @tariff_holder.meters.gas
+      else
+        @meters = []
+      end
+    end
+
+    def update_meters
+      if @energy_tariff.update(energy_tariff_params.merge(updated_by: current_user))
+        redirect_to energy_tariffs_path(@energy_tariff)
+      else
+        render :edit_meters
+      end
+    end
+
+    def edit_dates
+    end
+
+    def update_dates
```

Comment:
Rather than a separate route, I'd just send the user back to the user title / date editing form. No need to exclude editing of titles, that's a useful feature in itself

Reply:
:+1: